<launch>

<!-- =================================================================================
                      Launch Task Management Stuff
     ================================================================================= -->

<arg name="smach_viewer" doc="(bool) bring up the smach viewer"/>
<arg name="bringup_arm1" doc="(bool) launch arm1"/>

<!-- startup script to make sure the arms are running the right controllers -->
<node name="peg_demo_startup"
  pkg="peg_assist_demo"
  type="startup.py" />

<!-- view execution progress -->
<node if="$(arg smach_viewer)"
  name="smach_viewer"
  pkg="smach_viewer"
  type="smach_viewer.py" />

<!-- publish planning information -->
<node name="planning_scene_script" pkg="peg_assist_demo" type="publish_planning_scene.py" output="screen">
  <param name="wait_for" value="/gazebo/w2barrett_manager/move_group"/>
  <param name="service" value="/gazebo/wam2/publish_planning_scene"/>
  <param name="wait_time" value="0.0"/>
  <param name="rate" value="0.1"/>
</node>

<node name="planning_scene_script_arm1" pkg="peg_assist_demo" type="publish_planning_scene.py" output="screen"
  if="$(arg bringup_arm1)">
  <param name="wait_for" value="/gazebo/barrett_manager/move_group"/>
  <param name="service" value="/gazebo/wam/publish_planning_scene"/>
  <param name="wait_time" value="0.0"/>
  <param name="rate" value="0.1"/>
</node>

<!--<node name="task_script" type="iros2014.py" pkg="peg_assist_demo"/>-->
</launch>
